standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -13317,6 +13325,10 @@ public int markScheduledExecutionsTimedOut(int timeoutSecs) throws InvalidOperat
       recoverInvalidScheduledQueryState(timeoutSecs);
       committed = commitTransaction();
       return results.size();
+    } catch (InvalidOperationException | MetaException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new MetaException(e.getMessage());
```

Comment:
are these catches needed?

Reply:
yes, in my ide shows ```Unhandled exception from auto-closeable resource: java.lang.Exception``` without the catch clause ```catch (Exception e)``` and fails compilation.  ```try (QueryWrapper wrapper = new QueryWrapper())``` works ok here, but ```QueryWrapper``` may be going to be removed later.  I trying to replace with ```QueryWrapper```.

Reply:
I don't know what kind of IDE you are using; but mine(eclipse) and maven is perfectly fine with this:
```
  public static void main(String[] args) {
    try (aa x = new aa()) {
      System.out.println("body");
    } finally {
      System.out.println("final");
    }
  }

  static class aa implements AutoCloseable {
    public aa() {
      throw new RuntimeException();
    }
    @Override
    public void close() {

    }
  }

```
note that `aa#close` doesn't throw `Exception` while `AutoCloseable#close` may.

but anyway - let's go back to to the try-with-resources free version; the repackaging of the exceptions is one thing(which happens too much; and sometimes incorrectly around objectstore) - but now that try-with-resources is used the the "opentransaction" is also inside the try block - if the try fails the finally will still run; and may make the "counter" off by one... - the best would be to rewrite that as well to use try-with-resources but I'm aftraid that will be start a cascade of changes....it seems there is already some places which are using `openTransaction` in an interesting fashion...

sorry for the detour; but its hard to make this part better..

Reply:
Agreed, I will update the codes lately. Thanks so much for your feedback, @kgyrtkirk!